File: app/src/main/java/com/neome/java/api/meta/base/dto/DefnStudioPickFieldId.java
// neome.ai API - do not change
//

package com.neome.java.api.meta.base.dto;

import com.neome.java.api.meta.base.Types.EnumDefnCompType;
import com.neome.java.api.meta.base.Types.MetaIdComposite;
import com.neome.java.api.meta.base.Types.MetaIdField;
import com.neome.java.api.meta.base.Types.MetaIdForm;

import org.jetbrains.annotations.Nullable;

@SuppressWarnings("unused")
public class DefnStudioPickFieldId extends DefnFieldEditable
{
  @Nullable
  public MetaIdComposite[] compositeIdSet;

  @Nullable
  public MetaIdField[] excludeFieldIdSet;

  @Nullable
  public EnumDefnCompType[] filterFieldTypeSet;

  public MetaIdForm formId;

  @Nullable
  public MetaIdField[] includeFieldIdSet;

  @Nullable
  public DefnStudioMapOfDtoOption includeOptionMap;

  @Nullable
  public Boolean showCompositeName;
}

File: app/src/main/java/com/neome/api/meta/base/dto/DefnStudioPickFieldId.kt
// neome.ai API - do not change
//
@file:Suppress("unused", "MemberVisibilityCanBePrivate", "CanBeParameter", "PropertyName")


package com.neome.api.meta.base.dto

import com.neome.api.meta.base.Types.EnumDefnCompType
import com.neome.api.meta.base.Types.MetaIdComposite
import com.neome.api.meta.base.Types.MetaIdField
import com.neome.api.meta.base.Types.MetaIdForm

interface DefnStudioPickFieldId : DefnFieldEditable {
    val compositeIdSet: List<MetaIdComposite>?
    val excludeFieldIdSet: List<MetaIdField>?
    val filterFieldTypeSet: List<EnumDefnCompType>?
    val formId: MetaIdForm
    val includeFieldIdSet: List<MetaIdField>?
    val includeOptionMap: DefnStudioMapOfDtoOption?
    val showCompositeName: Boolean?
}
